### PR TITLE
Add linker build for the toolchain installation

### PR DESCRIPTION
**Pull request by akiramenai**
Also remove code duplication in driver.

### CODE REVIEW

File: llvm/tools/clang/lib/Driver/ToolChains/TVM.cpp [hunk 1]
```diff
@@ -23,14 +23,16 @@ using namespace clang::driver::toolchains;
 using namespace clang;
 using namespace llvm::opt;
 
+bool tvm::isTVMArch(StringRef Str) { return Str.find("tvm") == 0; }
```
Comment (azhogin) [inline llvm/tools/clang/lib/Driver/ToolChains/TVM.cpp:26]: Are you sure that `find("tvm")` is enough?

Comment (akiramenai) [reply llvm/tools/clang/lib/Driver/ToolChains/TVM.cpp:26]: The predicate is that the architecture starts with "tvm". At this time in the driver, we only have the triple as a string. We might have at least two sensible possibilities here: `tvm` and `tvm-unknown-unknown`. I also believe that `none` for OS is also OK, etc.
Probably we could check that either we have exactly tvm, or the triple starts with `tvm-`. It should make it impossible for other architecture occasionally starting with tvm to be confused with tvm unless it is called exactly the same.

Comment (azhogin) [reply llvm/tools/clang/lib/Driver/ToolChains/TVM.cpp:26]: As far as I understand, normal way to work with triple strings is using `Triple` helper class. Which will parse string and provide parsed info (`tr.getArch() == Triple::tvm`). Or add boolean getter like `Triple::isNVPTX`, `Triple::isARM`.
Looks like `Triple` object is already exists in `Compilation::getDefaultToolChain`.
An example of usage:
```
const ToolChain &TC = C.getDefaultToolChain();
  if (TC.getTriple().isOSBinFormatMachO()) ...
```

Comment (akiramenai) [reply llvm/tools/clang/lib/Driver/ToolChains/TVM.cpp]: Fixed

**Pull request merged**